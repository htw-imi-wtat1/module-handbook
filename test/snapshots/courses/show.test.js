const { app, request, Course, removeID } = require('../../commonJest')

const courseData = {
  semester: 3,
  required: 'required',
  code: 'B13_index_snapshot',
  name: 'Bildverarbeitung',
  mission: 'work on pictures',
  ects: 2,
  examination: 'exam',
  objectives: 'learn to transform pictures',
  contents: 'loops over pixels',
  prerequisites: 'info2',
  literature: 'some books',
  methods: 'P SL/Ãœ',
  skills_knowledge_understanding: 'a',
  skills_intellectual: 'b',
  skills_practical: 'c',
  skills_general: 'd'
}

describe('Course Show', function () {
  let courseID
  beforeEach(function (done) {
    Course.create(courseData)
      .then(created => {
        courseID = created.id
        done()
      })
      .catch(error => {
        // console.log('error caught: ' + error.message)
        done(error.message)
      })
  })
  it('renders course show correctly', (done) => {
    request(app)
      .get(`/courses/${courseID}`)
      .then((response) => {
        expect(removeID(response.text, courseID)).toMatchSnapshot()
        done()
      })
  })
})
