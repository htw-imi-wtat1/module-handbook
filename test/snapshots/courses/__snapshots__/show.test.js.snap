// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Show renders course show correctly 1`] = `
"<!DOCTYPE html>
<html>

<head>
	<meta charset=\\"utf-8\\">
	<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=0.8, maximum-scale=1\\">
	<title>Module Handbook</title>
	<link rel=\\"shortcut icon\\" type=\\"image/png\\" href=\\"/images/icon.png\\" />
	<link rel=\\"stylesheet\\" href=\\"/css/bootstrap.css\\">
	<link rel=\\"stylesheet\\" href=\\"/css/module_handbook.css\\">
	<script src=\\"/jquery.min.js\\"></script>
	<script src=\\"/popper.js\\"></script>
	<script src=\\"/js/bootstrap.min.js\\"></script>

</head>

<body>
<div class = \\"container\\">
	<div id=\\"nav\\">
		<div class=\\"col-sm nav-align\\">
			<h1 id=\\"title\\">Module Handbook</h1>
		</div>
		<nav class=\\"navbar navbar-expand-lg navbar-light bg-light\\">
    <a class=\\"navbar-brand\\" href=\\"/\\">Home</a>
    <button class=\\"navbar-toggler\\" type=\\"button\\" data-toggle=\\"collapse\\" data-target=\\"#navbarSupportedContent\\"
            aria-controls=\\"navbarSupportedContent\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\">
        <span class=\\"navbar-toggler-icon\\"></span>
    </button>

    <div class=\\"collapse navbar-collapse\\" id=\\"navbarSupportedContent\\">
        <ul class=\\"navbar-nav mr-auto\\">
            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/modules/list\\">Module List
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/modules/tabular\\">Module Table
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/courses\\">Courses
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/users\\">Users
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/users/new\\">Register
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/about\\">About
        
    </a>
</li>

            


                    
                       <li class=\\"nav-item\\">
                          <a  class=\\"nav-link\\" href =\\"/users/login\\">Log In</a>
                       </li>
                    

        </ul>

    </div>
</nav>

	</div>
	<div class = \\"container\\">
		
		<div class = \\"flashes mt-4\\">
       
		</div>
	
	  <p>Course.</p>
<p>[&#34;code&#34;,&#34;name&#34;,&#34;ects&#34;,&#34;mission&#34;,&#34;examination&#34;,&#34;objectives&#34;,&#34;contents&#34;,&#34;prerequisites&#34;,&#34;literature&#34;,&#34;methods&#34;,&#34;skills_knowledge_understanding&#34;,&#34;skills_intellectual&#34;,&#34;skills_practical&#34;,&#34;skills_general&#34;]</p>
<h1>B13_index_snapshot - Bildverarbeitung</h1>
<p></p>
<table>
    
    <tr><td>code</td><td>B13_index_snapshot</td></tr>
    
    <tr><td>name</td><td>Bildverarbeitung</td></tr>
    
    <tr><td>ects</td><td>2</td></tr>
    
    <tr><td>mission</td><td>work on pictures</td></tr>
    
    <tr><td>examination</td><td>exam</td></tr>
    
    <tr><td>objectives</td><td>learn to transform pictures</td></tr>
    
    <tr><td>contents</td><td>loops over pixels</td></tr>
    
    <tr><td>prerequisites</td><td>info2</td></tr>
    
    <tr><td>literature</td><td>some books</td></tr>
    
    <tr><td>methods</td><td>P SL/Ãœ</td></tr>
    
    <tr><td>skills_knowledge_understanding</td><td>a</td></tr>
    
    <tr><td>skills_intellectual</td><td>b</td></tr>
    
    <tr><td>skills_practical</td><td>c</td></tr>
    
    <tr><td>skills_general</td><td>d</td></tr>
    
</table>
	</div>
</div>


</body>

</html>
"
`;
